ServerMonitoringSystem.MessageProcessor/Program.cs

```
@@ -10,42 +9,45 @@
 using ServerMonitoringSystem.Infrastructure.Repositories;
 using ServerMonitoringSystem.MessageProcessor.Persistence;
 
-var builder = new ConfigurationBuilder()
-    .SetBasePath(Directory.GetCurrentDirectory())
-    .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true);
-
-var configuration = builder.Build();
-
-var anomalyConfig = configuration
-    .GetSection("AnomalyDetectionConfig")
-    .Get<AnomalyDetectionConfig>();
+var anomalyConfig = new AnomalyDetectionConfig
+{
+    MemoryUsageAnomalyThresholdPercentage = double.Parse(Environment.GetEnvironmentVariable("MEMORY_ANOMALY_THRESHOLD") ?? "0.4"),
+    CpuUsageAnomalyThresholdPercentage = double.Parse(Environment.GetEnvironmentVariable("CPU_ANOMALY_THRESHOLD") ?? "0.5"),
+    MemoryUsageThresholdPercentage = double.Parse(Environment.GetEnvironmentVariable("MEMORY_USAGE_THRESHOLD") ?? "0.8"),
+    CpuUsageThresholdPercentage = double.Parse(Environment.GetEnvironmentVariable("CPU_USAGE_THRESHOLD") ?? "0.9")
+};
 
-var signalRConfig = configuration
-    .GetSection("SignalRConfig")
-    .Get<SignalRConfig>();
+var signalRConfig = new SignalRConfig
+{
+    SignalRUrl = Environment.GetEnvironmentVariable("SIGNALR_URL") ?? "https://localhost:7271/hub/alerts"
+};
 
-var mongoDbSettings = configuration
-    .GetSection("MongoDbSettings")
-    .Get<MongoDbSettings>();
+var MongoDbConnection = "mongodb+srv://mahmoudbmiqdad:mahmoud123@cluster0.bx8augs.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0";
```

these should be moved to the other configs

---

The connection string was already provided through the `.env` file, and the hardcoded value was only used as a fallback in case environment variables were missing. I've now updated the code to throw a clear exception instead, ensuring proper configuration is enforced.